### PR TITLE
Throw exception when misusing `error_to_string()`

### LINKED ISSUE #5356: `WP_CLI::error_to_string()` doesn't have a guaranteed return value
The method `WP_CLI::error_to_string()` is supposed to accept an error object and return an error message that can be output.

However, if you pass a type into the method that it doesn't recognize, it doesn't return anything at all.

This should never happen. If it does nevertheless, it should throw an `InvalidArgumentException` to make sure a developer sees this and fix the problem.

The exception could use a message like the following:
`Unsupported argument type passed to WP_CLI::error_to_string(): {$type}`

### PR DESCRIPTION
**Pull request by vsamoletov**
Fixes #5356

### CODE REVIEW

File: tests/test-wp-cli.php [hunk 1]
```diff
@@ -5,4 +5,9 @@ class WP_CLI_Test extends PHPUnit_Framework_TestCase {
 	public function testGetPHPBinary() {
 		$this->assertSame( WP_CLI\Utils\get_php_binary(), WP_CLI::get_php_binary() );
 	}
+
+	public function testErrorToString() {
+		$this->setExpectedException( 'InvalidArgumentException', "Unsupported argument type passed to WP_CLI::error_to_string(): 'boolean'" );
+		WP_CLI::error_to_string( (bool) true );
```
Comment (swissspidy) [inline tests/test-wp-cli.php:11]: No need for casting a boolean to a boolean

```suggestion
		WP_CLI::error_to_string( true );
```

Comment (vsamoletov) [reply tests/test-wp-cli.php:11]: Thanks.

**Comment by vsamoletov** [comment]
I have to add @throws to PHPDoc probably. Please let me know.

**Comment by vsamoletov** [comment]
@swissspidy could you please also review 68d96c9

**Pull request merged**